crates/driver/src/domain/competition/auction.rs

```diff
@@ -23,26 +34,52 @@ pub struct Auction {
 }
 
 impl Auction {
-    pub fn new(
+    pub async fn new(
         id: Option<Id>,
-        mut orders: Vec<competition::Order>,
+        orders: Vec<competition::Order>,
         tokens: impl Iterator<Item = Token>,
-        gas_price: eth::GasPrice,
         deadline: Deadline,
-        weth: eth::WethAddress,
-    ) -> Result<Self, InvalidTokens> {
+        eth: &Ethereum,
+    ) -> Result<Self, Error> {
         let tokens = Tokens(tokens.map(|token| (token.address, token)).collect());
 
         // Ensure that tokens are included for each order.
+        let weth = eth.contracts().weth_address();
         if !orders.iter().all(|order| {
             tokens.0.contains_key(&order.buy.token.wrap(weth))
                 && tokens.0.contains_key(&order.sell.token.wrap(weth))
         }) {
-            return Err(InvalidTokens);
+            return Err(Error::InvalidTokens);
         }
 
-        // Sort orders such that most likely to be fulfilled come first.
-        orders.sort_by_key(|order| {
+        Ok(Self {
+            id,
+            orders,
+            tokens,
+            gas_price: eth.gas_price().await?,
+            deadline,
+        })
+    }
+
+    /// [`None`] if this auction applies to a quote. See
+    /// [`crate::domain::quote`].
+    pub fn id(&self) -> Option<Id> {
```

Comment:
Separate, but solvers have been asking us to also send IDs for quote instances (so they can easily debug and get back to us with identifiable instances). One idea we had was to "pre-generate" the quote_id and then attach it to each request. This will even allow solvers to check independently what they quoted for a given order (assuming we expose quote_id in our API)

Reply:
I _think_ some solvers use the `id` field to disambiguate between a `solve` and a `quote` (I think Quasimodo does this specifically). This might be a backwards incompatible change (just warning).

I 100% agree we should do this, however.

Also note that this `id` is the id of an auction to `solve`. Driver's have a separate API for handling quotes.